### PR TITLE
[lldb] Use triple to check for arch

### PR DESCRIPTION
**Pull request by augusto2112**
rdar://86042195

### CODE REVIEW

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by augusto2112** [comment]
@swift-ci test

File: lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp [hunk 1]
```diff
@@ -59,24 +59,18 @@ static lldb::addr_t
...
-  } else if (core_kind >= ArchSpec::Core::kCore_arm_first &&
-             core_kind <= ArchSpec::Core::kCore_arm_last) {
+  if (triple.isAArch64() || triple.isARM()) {
     is_arm = true;
-  } else if (core_kind >= ArchSpec::Core::kCore_x86_64_first &&
-             core_kind <= ArchSpec::Core::kCore_x86_64_last) {
-    is_intel = true;
-  } else if (core_kind >= ArchSpec::Core::kCore_x86_32_first &&
-             core_kind <= ArchSpec::Core::kCore_x86_32_last) {
+  } else if (triple.isX86() || triple.isX32()) {
```
Comment (adrian-prantl) [inline lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp]: IIUC X32 (= x84_64 with 32-bit pointers) is not the same as kCore_x86_32 (= i386). I think this should just be `isX86()`.

**Review by adrian-prantl — APPROVED** [review]
Thanks, this is much better! 

Can you do the same thing with the function below this? Also the if statements should probably be else ifs.

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by augusto2112** [comment]
@swift-ci test

**Pull request merged**